funnel/models/proposal.py

```diff
@@ -164,7 +165,82 @@ def _validate_proposal_space(self, key, value):
                 redirect.proposal = self
         return value
 
-    @state.transition(None, state.DELETED)
+    @with_roles(call={'admin'})
+    # XXX: Added `type` metadata just to decide color on the front end button. Needed?
+    @state.transition(state.AWAITING_DETAILS, state.DRAFT, title='Draft it', type='danger')
+    def draft(self):
+        pass
+
+    @with_roles(call={'author', 'creator'})
+    @state.transition(state.DRAFT, state.SUBMITTED, type='success')
+    def submit(self):
+        pass
+
+    state.add_state_group('CONFIRMABLE', state.SUBMITTED, state.WAITLISTED, state.SHORTLISTED)
+
+    @with_roles(call={'admin'})
+    @state.transition(state.CONFIRMABLE, state.CONFIRMED, type='success')
+    def confirm(self):
+        pass
+
+    @with_roles(call={'admin'})
+    @state.transition(state.CONFIRMED, state.SUBMITTED, type='danger')
+    def unconfirm(self):
+        pass
+
+    state.add_state_group('WAITLISTABLE', state.SUBMITTED, state.CONFIRMED,
+                          state.SHORTLISTED, state.REJECTED, state.CANCELLED)
+
+    @with_roles(call={'admin'})
+    @state.transition(state.WAITLISTABLE, state.WAITLISTED, title='Put in waitlist', type='primary')
+    def waitlist(self):
+        pass
+
+    @with_roles(call={'admin'})
+    @state.transition(state.SUBMITTED, state.SHORTLISTED, type='success')
+    def shortlist(self):
+        pass
+
+    @with_roles(call={'admin'})
+    @state.transition(state.SUBMITTED, state.REJECTED, type='danger')
+    def reject(self):
+        pass
+
+    @with_roles(call={'admin'})
+    @state.transition(state.SUBMITTED, state.CANCELLED, type='danger')
+    def cancel(self):
+        pass
+
+    @with_roles(call={'admin'})
+    @state.transition(state.SUBMITTED, state.AWAITING_DETAILS, title='Wait for details', type='primary')
+    def awaiting_details(self):
+        pass
+
+    state.add_state_group('EVALUATEABLE', state.SUBMITTED, state.AWAITING_DETAILS)
+
+    @with_roles(call={'admin'})
+    @state.transition(state.EVALUATEABLE, state.UNDER_EVALUATION, title='Evaluate', type='success')
+    def under_evaluation(self):
+        pass
+
+    state.add_state_group('SHORLISTABLE', state.SUBMITTED,
+                          state.AWAITING_DETAILS, state.UNDER_EVALUATION)
+
+    @with_roles(call={'admin'})
+    @state.transition(state.SHORLISTABLE, state.SHORTLISTED_FOR_REHEARSAL, title='Shortlist for rehearsal', type='success')
+    def shortlist_for_rehearsal(self):
+        pass
+
+    @with_roles(call={'admin'})
+    @state.transition(state.SHORTLISTED_FOR_REHEARSAL, state.REHEARSAL, title='Let\'s rehearse!', type='success')
+    def rehearsal(self):
+        pass
+
+    state.add_state_group('DELETABLE', state.DRAFT, state.SUBMITTED)
```

Comment:
Can you move these `state.add_state_group` calls to the top (beneath the column names) and place them together, so it's easy to scan for which groups are available?